hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieBaseFileGroupRecordBuffer.java

```diff
@@ -147,6 +153,37 @@ public void close() {
     records.clear();
   }
 
+  /**
+   * Compares two {@link Comparable}s.  If both are numbers, converts them to {@link Long} for comparison.
+   * If one of the {@link Comparable}s is a String, assumes that both are String values for comparison.
+   *
+   * @param o1 {@link Comparable} object.
+   * @param o2 other {@link Comparable} object to compare to.
+   * @return comparison result.
+   */
+  @VisibleForTesting
+  static int compareTo(Comparable o1, Comparable o2) {
+    // TODO(HUDI-7848): fix the delete records to contain the correct ordering value type
+    //  so this util with the number comparison is not necessary.
+    try {
+      return o1.compareTo(o2);
+    } catch (ClassCastException e) {
+      if (o1 instanceof Number && o2 instanceof Number) {
+        Long o1LongValue = ((Number) o1).longValue();
+        Long o2LongValue = ((Number) o2).longValue();
+        return o1LongValue.compareTo(o2LongValue);
```

Comment:
can possibly lead to wrong result with float/double?

Reply:
We can constrain the comparison to Long and Integer only to limit the possibility of wrong results.  I'll create a follow-up PR to fix this.

Reply:
HUDI-7852 to track.